Userland/Libraries/LibJS/Runtime/PropertyName.h

```diff
@@ -43,14 +43,21 @@ class PropertyName {
 
     template<Integral T>
     PropertyName(T index)
-        : m_type(Type::Number)
-        , m_number(index)
     {
         // FIXME: Replace this with requires(IsUnsigned<T>)?
         //        Needs changes in various places using `int` (but not actually being in the negative range)
         VERIFY(index >= 0);
```

Comment:
This might be out of scope for your PR, but actually doing what this comments suggests will probably weed out more issues with using incorrect types for array indicies.

Reply:
I'll have a look what it finds

Reply:
Hmm not sure we want this as `PropertyName {1}` is no longer allowed and `PropertyName{0}` gets converted to a nullptr for char* I think

Reply:
It did find a number of places where we wrongly use `i32` so it might be something to look into

Reply:
yeah thats fine, you can leave it for a future PR